build.gradle

```diff
@@ -402,6 +408,41 @@ task run(type: Run) {
   impliesSubProjects = true
 }
 
+task wrapper(type: Wrapper)
+
+gradle.projectsEvaluated {
+
+  def wrapperClosure = { Wrapper wrapper ->
+    wrapper.distributionType = DistributionType.ALL
+
+    wrapper.doLast {
+      final DistributionLocator locator = new DistributionLocator()
+      def version = GradleVersion.version(wrapper.gradleVersion)
+      final URI uri =
+              locator.getDistributionFor(version, wrapper.distributionType.name().toLowerCase(Locale.ENGLISH))
+      uri.toURL().withInputStream { i ->
+        final byte[] buffer = new byte[1 << 20]
+        final MessageDigest sha256 = MessageDigest.getInstance("SHA-256")
+        int bytesRead
+        while ((bytesRead = i.read(buffer)) >= 0) {
+          if (bytesRead > 0) {
+            sha256.update(buffer, 0, bytesRead)
```

Comment:
maybe simpler to just write 
```
String checksum = MessageDigest.getInstance("SHA-256").digest(uri.toURL().getBytes()).encodeHex().toString()
```

Reply:
It is simpler, but I would prefer not to download an arbitrary number of bytes into memory in one go. This is why I chunk it.